### PR TITLE
removed empty-expr and all of its dependencies

### PR DESCRIPTION
**Pull request by MajeedKazemi**

### CODE REVIEW

File: src/editor/validator.ts [hunk 1]
```diff
@@ -24,7 +23,6 @@ export class Validator {
         const context = providedContext ? providedContext : this.module.focus.getContext();
 
         return (
-            context.token instanceof EmptyExpr ||
```
Comment (MajeedKazemi) [inline src/editor/validator.ts:27]: This is part of the discussion about separating the Validator from the Type-Checker.

File: src/notification-system/error-msg-generator.ts [hunk 2]
```diff
@@ -233,12 +233,6 @@ export class ErrorMessageGenerator {
                                 CSSClasses.keyword
                             )} is a control flow statement that I can only put a range
                                 of values into. For this particular hole I should be using a variable name that is used to represent an element of the reange I am looping over.`;
-                        } else if (args.focusedNode instanceof EmptyExpr) {
```
Comment (MajeedKazemi) [inline src/notification-system/error-msg-generator.ts:236]: @vicchig Is this alright?

File: src/syntax-tree/ast.ts [hunk 3]
```diff
@@ -1447,10 +1463,10 @@ export class BinaryBoolOperatorExpr extends Expression {
 
         this.leftOperandIndex = this.tokens.length;
         this.tokens.push(new NonEditableTkn("(", this, this.tokens.length));
-        this.tokens.push(new EmptyExpr(this, this.tokens.length));
+        this.tokens.push(new TypedEmptyExpr([DataType.Any], this, this.tokens.length));
```
Comment (vicchig) [inline src/syntax-tree/ast.ts:1466]: Should this only accept booleans?

File: src/syntax-tree/ast.ts [hunk 4]
```diff
@@ -1447,10 +1463,10 @@ export class BinaryBoolOperatorExpr extends Expression {
...
         this.tokens.push(new NonEditableTkn(" " + operator + " ", this, this.tokens.length));
         this.rightOperandIndex = this.tokens.length;
-        this.tokens.push(new EmptyExpr(this, this.tokens.length));
+        this.tokens.push(new TypedEmptyExpr([DataType.Any], this, this.tokens.length));
```
Comment (vicchig) [inline src/syntax-tree/ast.ts:1469]: Same as above.

**Pull request merged**